### PR TITLE
support protobuf  serialization (#871)

### PR DESCRIPTION
**Pull request by leizhiyuan**
### Ⅰ. Describe what this PR did

for run the other test cases ，

```
    private static Configuration configuration = ConfigurationFactory.getInstance();

    private static String serialize = "protobuf";//configuration.getConfig(ConfigurationKeys.SERIALIZE_FOR_RPC);
    private final static String PROTOBUF = "protobuf";

```

I set serialize = "protobuf", normally, it will be set from config,  after review, I will change it.

the original transport models are more complex, there are many inheritance relationships, If you agree we need to keep simple for the transport object, I can change it to some flat models.

### Ⅱ. Does this pull request fix one issue?

fixes https://github.com/seata/seata/issues/823

### CODE REVIEW

**Comment by zhangthen** [comment]
@leizhiyuan   Add  javadoc  for the  new  codes.

File: config/seata-config-core/src/main/resources/file.conf [hunk 1]
```diff
@@ -19,6 +19,7 @@ transport {
     #auto default pin or 8
     worker-thread-size = 8
   }
+  serialize = "default"
```
[the remark this answers is not in the document]
Comment (leizhiyuan) [reply config/seata-config-core/src/main/resources/file.conf]: ok， I will use serialization

File: core/src/main/java/io/seata/core/protocol/HeartbeatMessage.java [hunk 2]
```diff
@@ -43,4 +43,12 @@ private HeartbeatMessage(boolean pingOrPong) {
     public String toString() {
         return this.pingOrPong ? "services ping" : "services pong";
     }
+
+    public boolean isPingOrPong() {
```
[the remark this answers is not in the document]
Comment (leizhiyuan) [reply core/src/main/java/io/seata/core/protocol/HeartbeatMessage.java]: this is the original code。

**Comment by leizhiyuan** [comment]
> @leizhiyuan Add javadoc for the new codes.

ok，I will pr later

**Comment by leizhiyuan** [comment]
> And can you add some test cases?

I have added some test cases ，you can review it

**Comment by zhangthen** [comment]
Now there are  two serialize-type: seata-diy and probuff,  you can add a enum class for serialize-types.

**Comment by xingfudeshi** [comment]
please update copyright to seata.

**Comment by lovepoem** [comment]
Please resolve the conflicts

**Pull request closed without merging**